src/cloudsc_pyiface/src/cloudscpytools/cloudsc_data.py

```diff
@@ -0,0 +1,457 @@
+            bfield=b2field
+        elif ldims == 3:
+            b3field=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype="float64")))
+            clsc.expand_mod.expand_r2(lfield, b3field,  nlon, nproma, nlev, ngptot, nblocks)
+            bfield=b3field
+        elif ldims == 4:
+            b4field=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype="float64")))
+#Note that we are using expand_r3bis here.
+#Original expand_r3 does not pass nlev from python to Fortran
+#            print ("This is 4D field")
+#            print ("lfield")
+#            print (lfield.shape)
+#            print ("bfield")
+#            print (b4field.shape)
+            clsc.expand_mod.expand_r3bis(lfield, b4field,  nlon, nproma, ndim,  ngptot, nblocks)
+            bfield=b4field
+        else:
+            print ("Wrong float ldim")
+    elif lfield.dtype == "bool":
+#Workaround - using type int32, otherwise complains about type disagreement at runtime
+        bfield=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype='int32')))
+        if ldims == 2:
+            tlfield=lfield.astype('int32')
+            clsc.expand_mod.expand_l1(tlfield, bfield,  nlon, nproma, ngptot, nblocks)
+        else:
+            print ("Wrong bool ldim")
+    elif lfield.dtype == "int32":
+        bfield=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype='int32')))
+        if ldims == 2:
+            clsc.expand_mod.expand_i1(lfield, bfield,  nlon, nproma, ngptot, nblocks)
+        else:
+            print ("Wrong int ldim")
+    else:
+        print ("Wrong dtype")
+    return bfield
+
+def load_input_fortran_fields(path, nparms, fields):
+    """
+    load_input_fortran_fields returns:
+    - set of variables needed to initiate computation of the Fortran kernel.
+    """
+    nproma =nparms['NPROMA']
+    nlev   =nparms['NLEV']
+    nblocks=nparms['NBLOCKS']
+    argnames_nlev = [
+        'pt', 'pq',
+        'pvfa', 'pvfl', 'pvfi', 'pdyna', 'pdynl', 'pdyni',
+        'phrsw', 'phrlw','pvervel','pap','plu','plude',
+        'psnde', 'pmfu', 'pmfd',
+        'pa', 'psupsat',
+        'plcrit_aer','picrit_aer','pre_ice',
+        'pccn', 'pnice'
+    ]
+    argnames_nlevp = [
+        'paph'
+    ]
+
+    argnames_withnclv= [
+        'pclv','tendency_tmp_cld'
+    ]
+
+    argnames_tend = [
+        'tendency_tmp_t','tendency_tmp_q','tendency_tmp_a'
+    ]
+
+    argnames_scalar = [
+         'kfldx'
+    ]
+
+    argnames_nproma = [
+        'plsm', 'ldcum', 'ktype'
+    ]
+
+
+    with h5py.File(path, 'r') as f:
+        fields['KLON'] = f['KLON'][0]
+        fields['KLEV'] = f['KLEV'][0]
+        fields['PTSPHY'] = f['PTSPHY'][0]
+        klon=fields['KLON']
+
+        for argname in argnames_nlev:
+            print('Loading field:',argname)
+            fields[argname] = field_c_to_fortran((nblocks,nlev,nproma),
+                                                 f[argname.upper()],nparms,klon)
+
+        for argname in argnames_nlevp:
```

Comment:
Grouping and iterating seems overkill here. Why not simply list them as 
```
fields['pt'] = ield_c_to_fortran(('PT', dims=(nblocks,nlev,nproma), ...)
```

Reply:
This solution seems much cleaner and portable to me. I am strongly in favour of my solution over the explicit list at the moment.

Reply:
Not sure what "portable" in this context means, but no matter. I can see your point if there's only one set, but of you have to group by dimension, I actually find this unnecessarily hard to read, due to the indirection forcing me to look in different places to find out which shape a field has. And inferring the shape (a tuple) from a naming convention is just plain annoying, but I guess this comes down to taste...